marketplaceItems/avatarCustomization101/appResources/appData/avatarCustomization101.js

```diff
@@ -0,0 +1,937 @@
+        STRING_FLOW = CONFIG.STRING_FLOW,
+        STRING_INFO = CONFIG.STRING_INFO,
+        STRING_STATE = CONFIG.STRING_STATE;
+
+    // UI variables
+    var ui;
+    var dynamicData = deepCopy(CONFIG.INITIAL_DYNAMIC_DATA);
+    // set default UI values to be parsed when set to defaults
+    var defaultMaterialProperties = {
+        shadeless: deepCopy(CONFIG.INITIAL_DYNAMIC_DATA[STRING_MATERIAL].shadeless),
+        pbr: deepCopy(CONFIG.INITIAL_DYNAMIC_DATA[STRING_MATERIAL].pbr),
+    }
+
+    // Tab dynamic variables
+    var currentTab;
+
+    function startup() {
+
+        ui = new AppUi({
+            buttonName: BUTTON_NAME,
+            home: URL,
+            onMessage: onMessage,
+            // graphicsDirectory: Script.resolvePath("./resources/icons/"),
+            onOpened: onOpened,
+            onClosed: onClosed
+        });
+
+        // check avatar, if avatar is Avi.fst (or fbx) then set APP_AVI_ENABLED state
+        // if not Avi avatar, dynamicData.aviEnabled is false
+        // loadAnimationsIntoCache();
+
+        Script.scriptEnding.connect(unload);
+    }
+
+    function onClosed() {
+
+        deleteMirror();
+        // save lastTab that the user was on
+        dynamicData.state.activeTabName = currentTab;
+
+        MyAvatar.hasScriptedBlendshapes = false;
+
+    }
+
+    function onOpened() {
+
+        if (DEBUG) {
+            print("ACA101 onOpened: isAviEnabled ", isAviYourCurrentAvatar());
+            print("ACA101 onOpened: activeTabName is ", dynamicData.state.activeTabName);
+        }
+
+        if (isAviYourCurrentAvatar()) {
+
+            setIsAviEnabledTrue();
+
+            // if your last closed tab has extra setup functionality
+            // ensure you have the correct view for the current tab
+            switchTabs(dynamicData.state.activeTabName);
+
+        } else {
+            setIsAviEnabledFalse();
+        }
+
+        updateUI();
+
+    }
+
+    function switchTabs(tabName) {
+        
+        var previousTab = currentTab;
+        currentTab = tabName;
+
+        // Flow tab conditionals
+        if (currentTab === STRING_FLOW && dynamicData[STRING_FLOW].showDebug){
+            // enable debug spheres
+            addRemoveFlowDebugSpheres(true);
+        }
+        if (previousTab === STRING_FLOW && currentTab !== STRING_FLOW){
+            // disable debug spheres
+            addRemoveFlowDebugSpheres(false);
+        }
+
+        // Blendshape tab conditionals
+        if(currentTab === STRING_BLENDSHAPES) {
+            // enable scripted blendshapes
+            MyAvatar.hasScriptedBlendshapes = true;
+            startBlendshapeInterval();
+        }
+        if (previousTab === STRING_BLENDSHAPES && currentTab !== STRING_BLENDSHAPES){
+            // disable scripted blendshapes
+            MyAvatar.hasScriptedBlendshapes = false;
+            stopBlendshapeInterval();
+        }
+
+    }
+
+    function unload() {
+
+
```

Comment:
please remove extra whitespace (i'm not making a comment on every instance of this i see)